argus/backends/azure/client.py

```diff
@@ -0,0 +1,288 @@
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import abc
+import random
+import six
+
+from argus import util
+from azure.common.credentials import UserPassCredentials
+from azure.mgmt.resource import ResourceManagementClient
+from azure.mgmt.storage import StorageManagementClient
+from azure.mgmt.network import NetworkManagementClient
+from azure.mgmt.compute import ComputeManagementClient
+
+PROVISIONING_STATE_CREATING = 'Creating'
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BaseAzureClient(object):
+    """Class for managing Azure instances
+
+    """
+    def __init__(self, subscription_id=None, username=None, password=None,
+                 resource_group_name=None, storage_account_name=None,
+                 availability_zone=None, image_vhd_path=None, vm_username=None,
+                 vm_password=None):
+        self.vm_disk_size = 50000
+        self._subscription_id = subscription_id
+        self._username = username
+        self._password = password
+        self._resource_group_name = resource_group_name
+        self._storage_account_name = storage_account_name
+        self._availability_zone = availability_zone
+        self._image_vhd_path = image_vhd_path
+        self._vm_username = vm_username
+        self.vm_password = vm_password
+        self._creds = UserPassCredentials(username, password)
+        self.res_client = ResourceManagementClient(
+            self._creds, subscription_id)
+        self._stor_client = StorageManagementClient(
+            self._creds, subscription_id)
+        self._vm_client = ComputeManagementClient(
+            self._creds, subscription_id)
+        self.net_client = NetworkManagementClient(
+            self._creds, subscription_id)
+
+    @abc.abstractmethod
+    def create_vm(self):
+        pass
+
+    @abc.abstractmethod
+    def destroy_vm(self):
+        pass
+
+    @abc.abstractmethod
+    def create_resource_group(self):
+        pass
+
+    @abc.abstractmethod
+    def create_storage_account(self):
+        pass
+
+    @abc.abstractmethod
+    def create_nic(self):
+        pass
+
+    @abc.abstractmethod
+    def get_floating_ip(self):
+        pass
+
+    @abc.abstractmethod
+    def get_vm_state(self):
+        pass
+
+
+class AzureClient(BaseAzureClient):
+    """Class for managing Azure instances
+    """
+    def __init__(self, subscription_id=None, username=None, password=None,
+                 resource_group_name=None, storage_account_name=None,
+                 availability_zone=None, image_vhd_path=None,
+                 vm_username=None, vm_password=None):
+        super(AzureClient, self).__init__(
+            subscription_id, username, password, resource_group_name,
+            storage_account_name, availability_zone, image_vhd_path,
+            vm_username, vm_password)
+
+        self.vm_name = "argusvm-" + AzureClient._get_random_id()
```

Comment:
We could use `util.rand_name`, also we could save this to a variable instead of calling `_get_random_id` each time.